core/src/main/scala/kafka/server/metadata/MetadataSnapshotter.scala

```diff
@@ -31,5 +32,5 @@ trait MetadataSnapshotter {
    *
    * @return                      True if we will write out a new snapshot; false otherwise.
    */
-  def maybeStartSnapshot(lastContainedLogTime: Long, image: MetadataImage): Boolean
+  def maybeStartSnapshot(lastContainedLogTime: Long, image: MetadataImage, reason: Set[SnapshotReason]): Boolean
```

Comment:
Let's add the new parameter to the documentation above.

Reply:
Added the `reason` as a `param` in the documentation.